### PR TITLE
feat(session): surface exec exit-status / exit-signal via SshSession.exitInfo

### PR DESCRIPTION
**Pull request by GlassOnTin**
## What

Adds `SshSession.exitInfo: Deferred<SessionExit?>` — the RFC 4254 §6.10 `exit-status` / `exit-signal` reports, which the packet loop previously parsed (the kaitai types were already wired into `SshMsgChannelRequest`) but dropped after logging. Without this, an exec consumer can read stdout/stderr to EOF but never learn how the command terminated.

- `SessionExit` sealed interface: `Status(code)` / `Signal(signalName, coreDumped, errorMessage)`
- Completed from the connection's channel-request dispatch for session channels; first report wins per the RFC
- Resolves to `null` when the channel closes without a report (servers aren't required to send one), so waiters are never left suspended
- Metalava signature updated; `metalavaCheckCompatibility` passes

## Why

Haven is adopting sshlib as its second SSH engine (context in GlassOnTin/Haven#58); the next slice is remote command execution, where automations branch on exit codes. This was the one missing piece for exec parity — happy to adjust the API shape (e.g. a suspend accessor instead of the exposed `Deferred`) if you'd prefer.

## Tests

- 4 `FakeSshServer` flow tests: exit-status completes `exitInfo`; exit-signal completes it; close-without-report resolves null; first report wins over a later one
- 2 Testcontainers integration tests against OpenSSH 9.9p2: `sh -c 'exit 42'` → `Status(42)`; a KILL-terminated command → `Signal("KILL", …)`
- Full `:sshlib:test` suite: 1170 tests, 0 failures (unit + integration, Docker available)

### CODE REVIEW

**Comment by GlassOnTin** [comment]
Some context on where this fits, @kruton, for whenever you next have review bandwidth (CI is green across JDK 17/21/25 + CodeQL).

This closes the last gap for adopting sshlib as a second SSH engine in **Haven** — the migration you kindly kicked off back in [GlassHaven/Haven#58](https://github.com/GlassHaven/Haven/issues/58) when you reached out about ssh-proto. Since then the exec (#95), subsystem (#98), SFTP (#112) and FIDO2/SK (#146) work all landed upstream, and Haven now has the per-engine building blocks built and tested against real servers:

- **SFTP** — shipped and device-verified on Haven's daily driver.
- **Shell / PTY** — done, contract-tested against JSch byte-for-byte (banner-first-bytes, echo, resize, clean-exit).
- **Port forwarding** (local / remote / dynamic SOCKS5) — done, contract-tested.

The one thing still missing is a truthful command exit status. Haven runs `execCommand` to detect tmux/zellij/screen sessions and to bootstrap Mosh/Eternal-Terminal, and those flows branch on the exit code — so the library dropping the RFC 4254 §6.10 `exit-status`/`exit-signal` report (the kaitai types already parse it; the packet loop just discards it after a debug log) means the exit code can't be surfaced honestly. Rather than fabricate a value, Haven has that whole path gated behind this PR: once it lands in a release, a sshlib-toggled Haven profile can run its terminal, tunnels and file transfer entirely over sshlib instead of JSch.

Happy to reshape the API if you'd prefer something other than an exposed `Deferred<SessionExit?>` — e.g. a `suspend fun awaitExit(): SessionExit?` accessor, or folding it into the existing close/lifecycle signalling. Whatever fits the library best; the wire parsing and tests are the substance.

File: sshlib/src/test/kotlin/org/connectbot/sshlib/client/FakeSshServer.kt [hunk 1]
```diff
@@ -779,6 +779,38 @@ class FakeSshServer(
         }
     }
 
+    /** SSH_MSG_CHANNEL_REQUEST `exit-status` (RFC 4254 section 6.10). */
+    suspend fun sendChannelExitStatus(recipientChannel: Int, exitStatus: Int) {
+        val payload = ByteArrayOutputStream()
```
Comment (kruton) [inline sshlib/src/test/kotlin/org/connectbot/sshlib/client/FakeSshServer.kt]: You should be able to use the `SshMsgChannelRequest`, `ChannelRequestExitStatus`, and `ChannelRequestExitSignal` types since they already support writing.

File: sshlib/src/test/kotlin/org/connectbot/sshlib/client/FakeSshServer.kt [hunk 2]
```diff
@@ -779,6 +779,38 @@ class FakeSshServer(
...
+    }
+
+    /** SSH_MSG_CHANNEL_REQUEST `exit-signal` (RFC 4254 section 6.10). */
+    suspend fun sendChannelExitSignal(
+        recipientChannel: Int,
+        signalName: String,
+        coreDumped: Boolean = false,
+        errorMessage: String = "",
+    ) {
+        val payload = ByteArrayOutputStream()
```
Comment (kruton) [inline sshlib/src/test/kotlin/org/connectbot/sshlib/client/FakeSshServer.kt]: You should be able to use the `SshMsgChannelRequest`, `ChannelRequestExitStatus`, and `ChannelRequestExitSignal` types since they already support writing.

File: sshlib/src/main/kotlin/org/connectbot/sshlib/client/SshConnection.kt [hunk 3]
```diff
@@ -2563,9 +2566,31 @@ class SshConnection(
...
                         }
                         val requestAccepted = when (val entry = channelRegistry.findByLocalRecipient(recipientChannel)) {
-                            is SshChannelRegistry.Entry.Established.Session -> entry.channel.receiveRequest(deliverRequest)
+                            is SshChannelRegistry.Entry.Established.Session -> entry.channel.receiveRequest {
+                                deliverRequest()
+                                // RFC 4254 section 6.10: surface how the remote
+                                // process terminated to exitInfo waiters.
+                                when (val fields = msg.requestSpecificFields()) {
+                                    is ChannelRequestExitStatus -> entry.channel.receiveExitInfo(
+                                        SessionExit.Status(fields.exitStatus().toInt()),
```
Comment (kruton) [inline sshlib/src/main/kotlin/org/connectbot/sshlib/client/SshConnection.kt]: It looks like the RFC says this should be `uint32`, but this is converting to integer here. Prefer `Long` to preserve the full range. It might be prudent to add a test for the 0xffff_ffffL boundary condition as well.

File: sshlib/src/test/kotlin/org/connectbot/sshlib/client/FakeSshServer.kt (hunk 2 above)
Comment (kruton) [reply sshlib/src/test/kotlin/org/connectbot/sshlib/client/FakeSshServer.kt]: Updated `sendChannelExitStatus` and `sendChannelExitSignal` in `FakeSshServer` to serialize packets using `SshMsgChannelRequest`, `ChannelRequestExitStatus`, and `ChannelRequestExitSignal`.

File: sshlib/src/main/kotlin/org/connectbot/sshlib/client/SshConnection.kt (hunk 3 above)
Comment (kruton) [reply sshlib/src/main/kotlin/org/connectbot/sshlib/client/SshConnection.kt]: Updated `SessionExit.Status.code` from `Int` to `Long` to preserve the full `uint32` range from RFC 4254 §6.10, updated `SshConnection` to pass `fields.exitStatus()` directly, and added a flow test for the `0xFFFF_FFFFL` boundary condition.

File: sshlib/src/test/kotlin/org/connectbot/sshlib/client/FakeSshServer.kt (hunk 1 above)
Comment (kruton) [reply sshlib/src/test/kotlin/org/connectbot/sshlib/client/FakeSshServer.kt]: Updated `sendChannelExitStatus` and `sendChannelExitSignal` in `FakeSshServer` to serialize packets using `SshMsgChannelRequest`, `ChannelRequestExitStatus`, and `ChannelRequestExitSignal`.
